c_src/pactffi_nif.c

```diff
@@ -474,6 +604,99 @@ static ERL_NIF_TERM with_query_parameter_v2(ErlNifEnv *env, int argc, const ERL_
     }
 }
 
+// Provider verifier functions
+
+static ERL_NIF_TERM verify_via_file(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[])
+{
+    if (!enif_is_binary(env, argv[0]))
+    {
+        return enif_make_badarg(env);
+    }
+    char *name = convert_erl_binary_to_c_string(env, argv[0]);
+    if (!enif_is_binary(env, argv[1]))
+    {
+        return enif_make_badarg(env);
+    }
+    char *scheme = convert_erl_binary_to_c_string(env, argv[1]);
+    if (!enif_is_binary(env, argv[2]))
+    {
+        return enif_make_badarg(env);
+    }
+    char *host = convert_erl_binary_to_c_string(env, argv[2]);
+    if (!enif_is_number(env, argv[3]))
+    {
+        return enif_make_badarg(env);
+    }
+    int port = convert_erl_int_to_c_int(env, argv[3]);
+    if (!enif_is_binary(env, argv[4]))
+    {
+        return enif_make_badarg(env);
+    }
+    char *path = convert_erl_binary_to_c_string(env, argv[4]);
+
+    if (!enif_is_binary(env, argv[5]))
+    {
+        return enif_make_badarg(env);
+    }
+    char *version = convert_erl_binary_to_c_string(env, argv[5]);
+
+    if (!enif_is_binary(env, argv[6]))
+    {
+        return enif_make_badarg(env);
+    }
+    char *branch = convert_erl_binary_to_c_string(env, argv[6]);
+
+    if (!enif_is_binary(env, argv[7]))
+    {
+        return enif_make_badarg(env);
+    }
+    char *file_path = convert_erl_binary_to_c_string(env, argv[7]);
+
+    if (!enif_is_binary(env, argv[8]))
+    {
+        return enif_make_badarg(env);
+    }
+    char *protocol = convert_erl_binary_to_c_string(env, argv[8]);
+
+    struct VerifierHandle *verifierhandle;
+    verifierhandle = pactffi_verifier_new_for_application(name, version);
+    pactffi_verifier_set_provider_info(verifierhandle, name, scheme, host, port, path);
+    pactffi_verifier_add_provider_transport(verifierhandle, protocol, port, path, scheme);
+    if (!enif_is_binary(env, argv[10]))
+    {
+        return enif_make_badarg(env);
+    }
+
+    char *state_path = convert_erl_binary_to_c_string(env, argv[10]);
+    if (state_path[0] != '\0')
+    {
+        pactffi_verifier_set_provider_state(verifierhandle, state_path, 0, 1);
+    }
+    pactffi_verifier_set_verification_options(verifierhandle, 0, 5000),
+    pactffi_verifier_set_publish_options(verifierhandle, version, NULL, NULL, -1, branch);
+    pactffi_verifier_add_directory_source(verifierhandle, file_path);
+    setenv("PACT_DO_NOT_TRACK", "true", 1);
+    int output = pactffi_verifier_execute(verifierhandle);
+    pactffi_verifier_shutdown(verifierhandle);
```

Comment:
👋 Hello, I'm interested in using this library and the functionality in this PR looks great.

For my use case, I'd want to pass in some headers and call `pactffi_verifier_add_custom_header` here.
What do you think about adding that as an extra parameter?

Reply:
Yeah, I am just still playing around with the functionality right now. But I think I will add the functionality when the existing verifier function easily hides the api call part for executing the provider message generator function.

Reply:
@rohan-b99 can you please raise an issue for this, I think this PR is already large enough. I will incorporate this request in another PR.